backend/ccm/canvas_api/section_api_handler.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from http import HTTPStatus
+from rest_framework.views import APIView
+from rest_framework import authentication, permissions
+from rest_framework.response import Response
+from rest_framework.request import Request
+
+from canvasapi.exceptions import CanvasException
+from canvasapi.course import Course
+from canvasapi import Canvas
+
+from backend.ccm.canvas_api.canvasapi_serializer import AddSectionInputSerializer, SectionSerializer
+from .exceptions import CanvasHTTPError
+from canvas_oauth.exceptions import InvalidOAuthReturnError
+
+from backend.ccm.canvas_api.canvas_credential_manager import CanvasCredentialManager
+
+from drf_spectacular.utils import extend_schema
+from rest_framework_tracking.mixins import LoggingMixin
+
+logger = logging.getLogger(__name__)
+
+CANVAS_CREDENTIALS = CanvasCredentialManager()
+
+class CanvasSectionAPIHandler(LoggingMixin, APIView):
+    logging_methods = ['GET', 'POST']
+    """"
+    "API handler for Canvas section data."
+    """
+    authentication_classes = [authentication.SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = SectionSerializer  # Ensures Swagger UI recognizes it
+
+    def get(self, request: Request, course_id: int) -> Response:
+        """
+        Get section data from Canvas.
+        """
+        try:
+            canvas_api: Canvas = CANVAS_CREDENTIALS.get_canvasapi_instance(request)
+            logger.info(f"Retrieving course for section data with course_id: {course_id}")
+            course: Course = canvas_api.get_course(course_id)
+            logger.info(f"Course data retrieved: {course}")
+            
+            # Call the Canvas API package to get section details.
+            logger.info(f"Getting section data for course")
+            # Get list of sections, including total_students info
+            sections = course.get_sections(include=['total_students'])
+            logger.info(f"Section data retrieved: {sections}")
+            
+            # Format the section data to return specific section info
+            formatted_sections = []
+            for section in sections:
+                formatted_sections.append({
```

Comment:
I would like this to be data class

from dataclasses import dataclass


@dataclass
class Section:
    id: int
    name: str
    course_id: int
    nonxlist_course_id: int
    total_students: int = 0
    
    this way this object can be reused. I feel GET/PUT Course can be adopt a similar approach but missed in that PR